src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala

```diff
@@ -24,6 +24,25 @@ trait InlineErrorChecker {
     recursivePreds
   }
 
+  /**
+    * Construct a call-graph of predicates specified by the given predicate ids. If any predicates
+    * are found to be mutually-recursive, print a warning to the console that it shall not be inlined.
+    * Return a set of predicates that are mutually-recursive.
+    *
+    * @param predicateIds the ids of the predicates we want to inline.
+    * @param program the program for which we are performing predicate inlining on.
+    * @return the set of mutually-recursive predicates.
+    */
+  def checkMutualRecursive(predicateIds: Set[String], program: Program): Set[Predicate] = {
+    val predicatesToInspect = predicateIds.map(program.findPredicate)
+    val predicateCallGraph = PredicateCallGraph.graph(predicatesToInspect, program)
+    val mutRecPreds = PredicateCallGraph.mutuallyRecursivePreds(predicateCallGraph)
+    mutRecPreds.foreach {
+      case Predicate(name, _, _) => println(s"Predicate: `$name` is mutually-recursive. Will not be inlined")
```

Comment:
I think the error message should be something like "$name is a recursive predicate and will not be inlined.", or even better, "$names are mutually recursive predicates and will not be inlined.", where `$names` is a list of the predicates involved in the mutual recursion cycle with `$name`
(You could even print out a little graph like `pred1 → pred2 → pred3 → pred1` but that's not important if the call graph is hard to manipulate)